### PR TITLE
add io/ioutil test

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:
I created the unit test for `internal/io/ioutil.go`
Also, I added the comment for each function of `internal/io/ioutil.go`

**NOTE**
It is too hard (maybe unreachable) to generate the error about `r.ReadFrom(f)`.
I ignored the test case about it, and also got SET WG agreement about it.

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/io/ioutil/ioutil_test.go [hunk 1]
```diff
@@ -18,13 +18,37 @@
...
+	fp, err := os.Create(path)
+	if err != nil {
+		log.Error(err)
+	}
+	err = fp.Chmod(0333)
+	if err != nil {
+		log.Error(err)
+	}
+	defer fp.Close()
+}
```
Comment (hlts2) [inline internal/io/ioutil/ioutil_test.go:50]: should we fatal when error occurs ? 🤔 
```suggestion
func genNonPermittedFile(t *testing.T, path string) {
	t.Helper()
	fp, err := os.Create(path)
	if err != nil {
		t.Fatal(err)
	}
	err = fp.Chmod(0333)
	if err != nil {
		t.Fatal(err)
	}
	defer fp.Close()
}
```

File: internal/io/ioutil/ioutil_test.go [hunk 2]
```diff
@@ -52,38 +76,124 @@ func TestReadFile(t *testing.T) {
...
+				name: "return ([]byte, nil error) when path is exist file name and file is empty",
+				args: args{
+					path: fName,
+				},
+				want: want{
+					want: []byte{},
+				},
+				beforeFunc: func(args) {
+					fp, err := os.Create(fName)
+					if err != nil {
```
Comment (hlts2) [inline internal/io/ioutil/ioutil_test.go:91]: Should we fatal when error occurs? 🤔

File: internal/io/ioutil/ioutil_test.go (hunk 1 above)
Comment (vankichi) [reply internal/io/ioutil/ioutil_test.go:50]: maybe, it is more safety 👍

**Comment by vankichi** [comment]
This PR will be closed because of  the deplicated of io/ioutil package in Go 1.16

**Pull request closed without merging**